cmake/common/targets/clang-tidy.cmake

```diff
@@ -1,130 +1,8 @@
-  set(CLANG_TIDY_BINARY_NAME_VAR "CLANG_TIDY_BINARY_NAME_${CLANG_TIDY_BINARY_SYSTEM}_${CLANG_TIDY_BINARY_ARCH}")
-  if(NOT DEFINED ${CLANG_TIDY_BINARY_URL_VAR} OR "${${CLANG_TIDY_BINARY_URL_VAR}}" STREQUAL "")
-    message(WARNING "Skipping `clang-tidy` download. No known pre-build binary URL")
-    return()
-  elseif(NOT DEFINED ${CLANG_TIDY_BINARY_CHECKSUM_VAR} OR "${${CLANG_TIDY_BINARY_CHECKSUM_VAR}}" STREQUAL "")
-    message(FATAL_ERROR "No known `clang-tidy` pre-build binary checksum")
-  elseif(NOT DEFINED ${CLANG_TIDY_BINARY_NAME_VAR} OR "${${CLANG_TIDY_BINARY_NAME_VAR}}" STREQUAL "")
-    message(FATAL_ERROR "No known `clang-tidy` pre-build binary name")
-  endif()
-  set(CLANG_TIDY_BINARY_URL "https://files.pythonhosted.org/packages/${${CLANG_TIDY_BINARY_CHECKSUM_VAR}}/${${CLANG_TIDY_BINARY_URL_VAR}}")
-
-  # Download and extract the pre-built binary ZIP if needed
-  set(CLANG_TIDY_BINARY_NAME "${${CLANG_TIDY_BINARY_NAME_VAR}}")
-  set(CLANG_TIDY_BINARY_OUTPUT "${SOURCEMETA_TARGET_CLANG_TIDY_ATTEMPT_INSTALL_OUTPUT_DIRECTORY}/${CLANG_TIDY_BINARY_NAME}")
-  if(EXISTS "${CLANG_TIDY_BINARY_OUTPUT}")
-    message(STATUS "Found existing `clang-tidy` pre-built binary at ${CLANG_TIDY_BINARY_OUTPUT}")
-    return()
-  endif()
-  set(CLANG_TIDY_BINARY_DOWNLOAD_DIR "${CMAKE_CURRENT_BINARY_DIR}/clang-tidy")
-  file(REMOVE_RECURSE "${CLANG_TIDY_BINARY_DOWNLOAD_DIR}")
-  file(MAKE_DIRECTORY "${CLANG_TIDY_BINARY_DOWNLOAD_DIR}")
-  set(CLANG_TIDY_BINARY_WHEEL "${CLANG_TIDY_BINARY_DOWNLOAD_DIR}/clang-tidy.whl")
-  message(STATUS "Downloading `clang-tidy` pre-built binary from ${CLANG_TIDY_BINARY_URL}")
-  file(DOWNLOAD "${CLANG_TIDY_BINARY_URL}" "${CLANG_TIDY_BINARY_WHEEL}"
-    STATUS CLANG_TIDY_BINARY_DOWNLOAD_STATUS SHOW_PROGRESS TLS_VERIFY ON
-    LOG CLANG_TIDY_BINARY_DOWNLOAD_LOG)
-  list(GET CLANG_TIDY_BINARY_DOWNLOAD_STATUS 0 _code)
-  if(NOT _code EQUAL 0)
-    message(WARNING "Failed to download the `clang-tidy` pre-built binary")
-    message(WARNING "${CLANG_TIDY_BINARY_DOWNLOAD_LOG}")
-    file(REMOVE_RECURSE "${CLANG_TIDY_BINARY_DOWNLOAD_DIR}")
-    return()
-  endif()
-  set(CLANG_TIDY_BINARY_EXTRACT_DIR "${CLANG_TIDY_BINARY_DOWNLOAD_DIR}/extracted")
-  file(MAKE_DIRECTORY "${CLANG_TIDY_BINARY_EXTRACT_DIR}")
-  file(ARCHIVE_EXTRACT INPUT "${CLANG_TIDY_BINARY_WHEEL}" DESTINATION "${CLANG_TIDY_BINARY_EXTRACT_DIR}")
-
-  # Install the binary
-  file(MAKE_DIRECTORY "${SOURCEMETA_TARGET_CLANG_TIDY_ATTEMPT_INSTALL_OUTPUT_DIRECTORY}")
-  file(COPY "${CLANG_TIDY_BINARY_EXTRACT_DIR}/clang_tidy/data/bin/${CLANG_TIDY_BINARY_NAME}"
-       DESTINATION "${SOURCEMETA_TARGET_CLANG_TIDY_ATTEMPT_INSTALL_OUTPUT_DIRECTORY}")
-  file(CHMOD "${CLANG_TIDY_BINARY_OUTPUT}" PERMISSIONS
-       OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
-  message(STATUS "Installed `clang-tidy` pre-built binary to ${CLANG_TIDY_BINARY_OUTPUT}")
+function(sourcemeta_enable_clang_tidy)
+  find_program(CLANG_TIDY_BIN NAMES clang-tidy)
```

Comment:
We could still download ClangTidy on demand and set `CMAKE_CXX_CLANG_TIDY` to the downloaded files?

Reply:
Also keep in mind that this `find_program` invocation succeeds even if `clang-tidy` was not found. Maybe worth putting `REQUIRED`? Otherwise you might end up setting a non-sense `CMAKE_CXX_CLANG_TIDY`?